### DISCUSSION TITLE
nominatim.socket not found

### DISCUSSION THREAD

**Discussion by gsg-git** [osm-search/Nominatim#3202] [Q&A]
Hello!

I have in the beginning overlooked the Python frontend and when i want to set up the Nominatim service it doesnt find the socket of nominatim:
Failed to start nominatim.service: Unit nominatim.socket not found.
when i search for the file there is no such file on the system.
Version 4.3.0
Ubuntu server 22.4.3

In which part of the installation scholud the socket be generated?

Slowly i thin i need to send you a really BIG coffe bag for all the questions.

Thank you in advance!

**Comment by gsg-git** [comment]
I switched to Docker.

**Comment by tribudiyono93** [comment]
same with me when try to use python front end in the step run nominatim as systemd job.
running well when using php but the result differrent with `https://nominatim.openstreetmap.org/ui/search.html` since this one using python front end.

**Comment by tribudiyono93** [comment]
after doing some research finally i can figure out the problem. so you need to create file `/etc/systemd/system/nominatim.socket` with content :
```
[Unit]
Description= nominatim socket

[Socket]
ListenStream=/run/nominatim.sock

[Install]
WantedBy=sockets.target
```

and after that you can run this command : 
sudo systemctl daemon-reload
sudo systemctl start nominatim.socket
sudo systemctl enable nominatim.socket

so you can start you nominatim.service

**Reply by lonvia** [reply]
I've updated the [release docs](https://nominatim.org/release-docs/latest/admin/Deployment-Python/) accordingly.
